custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomFileValidator.kt

```diff
@@ -56,30 +55,12 @@ class CustomFileValidator @Inject constructor(private val context: Context) {
   }
 
   @Suppress("NestedBlockDepth", "MagicNumber")
-  private fun getFileFromPlayAssetDelivery(): File? {
-    var zimFile: File? = null
+  private fun getFileFromPlayAssetDelivery(): FileDescriptor? {
     try {
       val context = context.createPackageContext(context.packageName, 0)
       val assetManager = context.assets
-      val inputStream = assetManager.open(BuildConfig.PLAY_ASSET_FILE)
-      val filePath = ContextCompat.getExternalFilesDirs(context, null)[0]
-      zimFile = File(filePath, BuildConfig.PLAY_ASSET_FILE)
-      // Write zim file data if file does not exist or corrupted
-      if (!zimFile.isFileExist() || zimFile.length() == 0L) {
-        // Delete previously corrupted file
-        if (zimFile.isFileExist()) zimFile.delete()
-        zimFile.createNewFile()
-        FileOutputStream(zimFile).use { outputSteam ->
-          inputStream.use { inputStream ->
-            val buffer = ByteArray(1024)
-            var length: Int
-            while (inputStream.read(buffer).also { length = it } > 0) {
-              outputSteam.write(buffer, 0, length)
-            }
-            outputSteam.flush()
-          }
-        }
-      }
+      val inputStream = assetManager.openFd(BuildConfig.PLAY_ASSET_FILE)
```

Comment:
[openFd](https://developer.android.com/reference/android/content/res/AssetManager#openFd(java.lang.String)) returns a [AssetFileDescriptor](https://developer.android.com/reference/android/content/res/AssetFileDescriptor) which is AutoCloseable.

So I suspect (to verify) that when we exit the function, `inputStream` is closed and so the fileDescriptor become invalid.
(You can check that with the [valid](https://developer.android.com/reference/java/io/FileDescriptor#valid()) function before passing it to `Archive`.)

It seems you can use [getParcelFileDescriptor](https://developer.android.com/reference/android/content/res/AssetFileDescriptor#getParcelFileDescriptor()) which return [ParcelFileDescriptor](https://developer.android.com/reference/android/os/ParcelFileDescriptor) which is also AutoCloseable.
So with some luck the ParcelFileDescriptor keep the fd open where FileDescriptor doesn't.